definition-src/Game/LambdaHack/Core/Random.hs

```diff
@@ -61,7 +62,7 @@ frequency :: Show a => Frequency a -> Rnd a
 frequency = St.state . rollFreq
 
 -- | Randomly choose an item according to the distribution.
-rollFreq :: Show a => Frequency a -> R.StdGen -> (a, R.StdGen)
+rollFreq :: Show a => Frequency a -> SM.SMGen -> (a, SM.SMGen)
```

Comment:
instead of `R.randomR (1, sumf)` I'd use `SM.nextWord32` and `` `mod` sumf`, it will remove a lot of code in a "hot loop".

If you stear at `randomIvalInteger` long enough, that's what it's doing anyway.

Reply:
Right. If also fusses around with magnitudes and wastes a lot of time calling a recursive function, even if only twice, but I can ignore the magnitudes, I think. First, I don't care for perfect randomness; artifacts may even introduce interesting patterns. Second, most of the time the ranges are quite small.

I would gladly use `randomS`, for Simple, implemented just like you describe, everywhere in my code and get rid (almost?) completely of `System.Random`.

Reply:
For the record: https://gist.github.com/phadej/6552b040ecdc342ae516e83daaaee1b0

Reply:
Seriously, `random` sucks so hard. BTW, is `bitmaskWithRejection` actually fastest? I'd guess `mod` is expensive, but `intMult` on amd64 should be really fast. Anyway, down with `randomR`.